piff/gsobject_model.py

```diff
@@ -289,27 +337,60 @@ def fit(self, star, fastfit=None, logger=None, convert_func=None):
         # these parameters don't really apply to each star separately.
         # After refluxing, we may drop this by 1 or 3 if adjusting flux and/or centroid.
         dof = np.count_nonzero(star.weight.array)
-        fit = star.fit.withNew(params=params, params_var=params_var,
-                               chisq=chisq, dof=dof)
+        fit = star.fit.newParams(params, params_var=params_var, num=self._num,
+                                 chisq=chisq, dof=dof)
         return Star(star.data, fit)
 
-    def initialize(self, star, logger=None):
+    def initialize(self, star, logger=None, default_init=None):
         """Initialize the given star's fit parameters.
 
-        :param star:  The Star to initialize.
-        :param logger:  A logger object for logging debug info. [default: None]
+        :param star:            The Star to initialize.
+        :param logger:          A logger object for logging debug info. [default: None]
+        :param default_init:    The default initilization method if the user doesn't specify one.
+                                [default: None]
 
         :returns: a new initialized Star.
         """
-        if self._centered:
-            params = np.array([ 1.0, 0.0, 0.0])
-            params_var = np.array([ 0.0, 0.0, 0.0])
+        logger = galsim.config.LoggerWrapper(logger)
+        init = self._init if self._init is not None else default_init
+        if init is None: init = 'hsm'
+        logger.debug("initializing GSObject with method %s",init)
+
+        flux, cenu, cenv, size, g1, g2, flag = star.hsm
+        if not self._raw_size > 0:
+            raise RuntimeError("Error computing the size of the reference object")
+        size /= self._raw_size
+
+        flux_scaling = None
+        if init == 'zero':
+            flux_scaling = 1.e-6
+        elif init == 'delta':
+            size *= 1.e-6
+        elif init.startswith('(') and init.endswith(')'):
```

Comment:
From the docstring, I expected this option to be a literal tuple rather than a stringified tuple.  Is there a reason this should be a string?

Reply:
I believe YAML doesn't recognize tuples, so (0.2, 0.4) gets parsed as a string.  But I'll add a real tuple option as well for people writing this directly from python.